serve/graph/model/block.go

```diff
@@ -37,6 +44,96 @@ func (b *Block) Time() time.Time {
 	return b.b.Time
 }
 
+func (b *Block) NumTxs() int64 {
+	return b.b.NumTxs
+}
+
+func (b *Block) TotalTxs() int64 {
+	return b.b.TotalTxs
+}
+
+func (b *Block) AppVersion() string {
+	return b.b.AppVersion
+}
+
+func (b *Block) LastBlockHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.LastBlockID.Hash)
+}
+
 func (b *Block) ProposerAddressRaw() string {
 	return b.b.ProposerAddress.String()
 }
+
+func (b *Block) LastCommitHash() string {
+	if b.b.LastCommitHash == nil {
+		return ""
+	}
+
+	return base64.StdEncoding.EncodeToString(b.b.LastCommitHash)
+}
+
+func (b *Block) ValidatorsHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.ValidatorsHash)
+}
+
+func (b *Block) NextValidatorsHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.NextValidatorsHash)
+}
+
+func (b *Block) ConsensusHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.ConsensusHash)
+}
+
+func (b *Block) AppHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.AppHash)
+}
+
+func (b *Block) LastResultsHash() string {
+	return base64.StdEncoding.EncodeToString(b.b.LastResultsHash)
+}
+
+func (b *Block) Txs() []*BlockTransaction {
+	return b.getTxs()
+}
+
+func (b *Block) getTxs() []*BlockTransaction {
+	// The function to unmarshal a block's transactions is executed once.
+	unmarshalTxs := func() {
+		var blockTxs []*BlockTransaction
+
+		for _, tx := range b.b.Txs {
+			blockTx := NewBlockTransaction(tx)
+			if blockTx != nil {
+				blockTxs = append(blockTxs, blockTx)
+			}
+		}
+
+		b.mu.Lock()
```

Comment:
this mutex is not needed because this will be called exactly once and it will block all `Do` calls.

Reply:
Fixed in this commit. https://github.com/gnolang/tx-indexer/pull/56/commits/4caff279e9ef12125d5e5c0fa707e9d32313c767